### ISSUE TITLE
reading case sensitive keys from yaml 

### ISSUE THREAD

**Issue by dani-t** [micronaut-projects/micronaut-core#2218]
The following issue appears when adding a property to a yaml configuration. I am going to provide the full example

### Steps to Reproduce

Currently the hibernate configuration is in hibernate.properties:

```
hibernate.dialect=org.hibernate.dialect.PostgreSQLDialect
hibernate.multiTenancy=SCHEMA
hibernate.tenant_identifier_resolver=com.....CurrentTenantIdentifierResolverImpl
hibernate.multi_tenant_connection_provider=com.....TenantConnectionProvider
```

When moving this config to yaml:

```
jpa:
  default:
    packages-to-scan:
      - '...'
    properties:
      hibernate:
        multiTenancy: SCHEMA
        multi_tenant_connection_provider: com.vdt.salv.domain.config.TenantConnectionProvider
        tenant_identifier_resolver: com.vdt.salv.domain.config.CurrentTenantIdentifierResolverImpl
        hbm2ddl:
          auto: none
```
If I keep the config only in the yaml file, at startup it will say that no multi tenancy strategy has been defined. If I move the multiTenancy config back to properties file, it works.

### Expected Behaviour

I would like to keep the configuration in only one place which is the yaml file. From what I know YAML keys should be case sensitive and so I would expect this not to be the problem. Actually I'm not really sure where is the problem but it looks like hibernate does not recognize that property or it is not passed as expected.

### Environment Information

- **Operating System**: Win10
- **Micronaut Version:** 1.2.0
- **JDK Version:** 12

**Comment by graemerocher** [comment]
Prior to closing this issue we need to modify https://github.com/micronaut-projects/micronaut-sql/blob/master/hibernate-jpa/src/main/java/io/micronaut/configuration/hibernate/jpa/JpaConfiguration.java#L140 to use `StringConvention.RAW` and release 1.2.3 of `micronaut-sql`

**Comment by graemerocher** [comment]
I have done this modification and released 1.2.3 of `micronaut-sql`. The PR needs to be merged before releasing Mircronaut 1.2.5
